internal/services/elastic/elastic_monitor_resource.go

```diff
@@ -0,0 +1,458 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceElasticMonitor() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceElasticMonitorCreate,
+		Read:   resourceElasticMonitorRead,
+		Update: resourceElasticMonitorUpdate,
+		Delete: resourceElasticMonitorDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ElasticMonitorID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ElasticMonitorName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"elastic_properties": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"elastic_cloud_user": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"email_address": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"id": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"elastic_cloud_sso_default_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+
+						"elastic_cloud_deployment": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Computed: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"name": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"deployment_id": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"azure_subscription_id": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"elasticsearch_region": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"elasticsearch_service_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"kibana_service_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+									"kibana_sso_url": {
+										Type:     pluginsdk.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"sku": {
```

Comment:
How about simplifying this to `sku_name`?

Reply:
It is how we are using in our api swagger.Also other similar plugins use the same format.This also gives us option to expand sku properties as it may not be just limited to name. Keeping it as is .